circle.cpp

```diff
@@ -0,0 +1,59 @@
+#include "circle.h"
+
+#include <glm/glm.hpp>
+
+#define _USE_MATH_DEFINES
+#include <math.h>
+
+#include <cmath>
```

Comment:
So, interesting issue with this. I only need these headers for `M_PI`, and in order to use that I need the `#define _USE_MATH_DEFINES` too. This works fine with the `<math.h>` header, but for `<cmath>` it seems it is made in a way where if anything is included before that macro is defined it will not work properly? The only way I could get it to work was by doing 
```
#define _USE_MATH_DEFINES

#include "circle.h"

#include <cmath>
```
Where the define had to come before even the circle header file to work, so it breaks the include order a bit. Doing this in CMake also worked. `target_compile_definitions(pong PRIVATE _USE_MATH_DEFINES)`
https://stackoverflow.com/questions/6563810/m-pi-works-with-math-h-but-not-with-cmath-in-visual-studio

Leaving it for now, but curious what you think,

Reply:
Very perplexing. Also hilariously these were finally added to c++ in c++20:

https://en.cppreference.com/w/cpp/numeric/constants

If you felt bold you could use c++20, I think MSVC has the best support of it of any of the major players. For the meantime though any of your solutions are totally fine, not worth stressing about